File: leetCode-swift/leetCode-swift/leetCode-0009.swift
//
//  leetCode-0009.swift
//  leetCode-swift
//
//  Created by liyoucheng on 2018/11/22.
//  Copyright © 2018年 Giga. All rights reserved.
//

import Foundation

func isPalindrome(_ x: Int) -> Bool {
    var y = x
    if (y < 0 || (y % 10 == 0 && y != 0)) {
        return false;
    }
    
    var revertedNumer = 0;
    while (y > revertedNumer) {
        revertedNumer = revertedNumer * 10 + y % 10;
        y /= 10;
    }
    
    return y == revertedNumer || y == revertedNumer / 10;
}

func test_0009() {
    let arr = [ 121, -121, 10 ]
    for num in arr {
        print(isPalindrome(num))
    }
}

File: leetCode-c/leetCode-c/LeetCode/leetCode-0009/leetCode-0009.c
//
//  leetCode-0009.c
//  leetCode-c
//
//  Created by liyoucheng on 2018/11/22.
//  Copyright © 2018年 Giga. All rights reserved.
//

#include "leetCode-0009.h"

#include <stdbool.h>

bool isPalindrome(int x) {
    if (x < 0 || (x % 10 == 0 && x != 0)) {
        return false;
    }
    
    int revertedNumer = 0;
    while (x > revertedNumer) {
        revertedNumer = revertedNumer * 10 + x % 10;
        x /= 10;
    }
    
    return x == revertedNumer || x == revertedNumer / 10;
}

void test_0009(void)
{
    int num[3] = { 121, -121, 10 };
    for (int i = 0; i < 3; i++) {
        bool ret = isPalindrome(num[i]);
        printf("ret = %d\n",ret);
    }
}
